mindmap/static/js/src/mindmap.js

```diff
@@ -39,11 +86,127 @@ function MindMapXBlock(runtime, element, context) {
+            const { assignments } = response;
+            $(element).find("#modal-submissions").addClass("modal_opened");
+           
+            showDataTable();
+
+            function showDataTable() {
+              const dataTableHeaderColumns = ["Username", "Uploaded", "Submitted", "Grade", "Actions"];
+              const dataTableHeaderColumnsHTML = dataTableHeaderColumns.reduce(
+                (prevColumn, currentColumn) => `${prevColumn}<th>${currentColumn}</th>`,
+                ""
+              );
+              const dataTableHTML = `
+                <table id="dataTable">
+                  <thead>
+                    <tr>
+                      ${dataTableHeaderColumnsHTML}
+                    </tr>
+                  </thead>
+                  <tbody>
+                    <!-- Data rows will be added here using DataTables -->
+                  </tbody>
+                </table>
+              `;
+
+              $(element).find(".modal__data").html(dataTableHTML);
+              $(element).find("#modal_title").html("Mindmap submissions");
+
+              const dataTable = $("#dataTable").DataTable({
+                data: assignments,
+                scrollY: "50vh",
+                dom: "Bfrtip",
+                bPaginate: false,
+                destroy: true,
+                columns: [
+                  { data: "username" },
+                  { data: "timestamp" },
+                  { data: "submitted" },
+                  { data: "score" },
+                  {
+                    data: null,
+                    render: function () {
+                      return '<button class="review_button button-link" type="button">Review</button>';
+                    },
+                  },
+                ],
+              });
+
+              handleRowDataTableClick(dataTable);
+            }
+
+            function handleRowDataTableClick(dataTable) {
+              $(element)
+                .find("#dataTable")
+                .on("click", ".review_button", function (e) {
+                  e.preventDefault();
+                  const target = $(e.target);
+                  const link = target;
+                  const rowReview = link.closest("tr");
+                  const submissionData = dataTable.row(rowReview).data();
+                  const answerMindMap = submissionData.answer_body.mindmap_student_body;
+                  const answerMindMapFormat = JSON.parse(answerMindMap);
+
+                  const mindMapReviewContainer = `
+                    <div class="review_mindmap_container">
+                      <button class="button-link back-review">&larr;&nbsp; Back</button>
+                      <div id="review-mindmap"></div>
+                      <div class="grade-assessment">
+                        <form id="grade-assessment-form">
+                          <div class="grade-assessment_form-control">
+                            <label for="grade">Grade</label>
```

Comment:
As I mentioned in my other comment, all titles/headers/...etc should be marked for translation